# Case Study on Testing of Hypothesis
### A company started to invest in digital marketing as a new way of their product promotions.For that they collected data and decided to carry out a study on it.
###   ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.
###   ● The company needs to check whether there is any dependency between the features “Region” and “Manager”.
### Help the company to carry out their study with the help of data provided.

In [11]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
data_set1=pd.read_csv('Sales_add.csv')
data_set1

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [12]:
data_set1.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

### 1. To check whether there is any increase in sales after stepping into digital marketing

In [13]:
print(np.var(data_set1['Sales_before_digital_add(in $)']), np.var(data_set1['Sales_After_digital_add(in $)']))

210329900.67975205 623460269.4710745


In [14]:
import scipy.stats as stats
stats.ttest_ind(a=data_set1['Sales_before_digital_add(in $)'], b=data_set1['Sales_After_digital_add(in $)'], equal_var=True)

Ttest_indResult(statistic=-12.995084451110877, pvalue=2.614368006904645e-16)

### Insights:
##### Here, the P value (= 2.614368006904645e-16) is much less than 0.05, we can assume that there is difference in sales after stepping to Digital Marketing. Thus we can reject Null Hypothesis

### 2. To check whether there is any dependency between the features “Region” and “Manager”

##### The dependency can be figured out by doing a hypothesis testing.If the testing results in a Null hypothesis,the given variables, Region and Manager will be independent of each other.
##### If the data counts for Alternate hypothesis, the variables will be dependent.

In [15]:
from scipy.stats import chi2_contingency
data_R = data_set1['Region']
data_M = data_set1['Manager']
data_crosstable= pd.crosstab(data_R,data_M)
data_crosstable

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [16]:
chiValue,pValue,dot,exp=chi2_contingency(data_crosstable)
chiValue,pValue,dot,exp

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

### Insights:
##### Here, the chi-square value obtained is 3.051 and the P-value is 0.5493991051158094, the P-value is much greater than 0.05, null hypothesis cannot be rejected.This means that no relationship exists on the categorical variables, Region and Manager; they are independent.